docs/tutorial/quick-start.md

```diff
@@ -1,22 +1,298 @@
+    └── out/my-electron-app-darwin-x64/my-electron-app.app/Contents/MacOS/my-electron-app
+    ```
+
+[node-download]: https://nodejs.org/en/download/
+
+## Learning the basics
+
+This section guides you through the basics of how Electron works under the hood. It aims at strengthening knowledge about Electron and the application created earlier in the Quickstart section.
+
+### Application architecture
+
+Electron consists of three main pillars:
+
+* **Chromium** for displaying web content.
+* **Node.js** for working with the local filesystem and the operating system.
+* **Custom APIs** for working with often-needed OS's native functions.
+
+Developing an application with Electron is like building a Node.js app with a web interface or building web pages with seamless Node.js integration.
+
+#### Main and Renderer Processes
+
+As it was mentioned before, Electron has two types of processes: Main and Renderer.
+
+* The Main process **creates** web pages by creating `BrowserWindow` instances. Each `BrowserWindow` instance runs the web page in its Renderer process. When a `BrowserWindow` instance is destroyed, the corresponding Renderer process gets terminated as well.
+* The Main process **manages** all web pages and their corresponding Renderer processes.
+
+----
+
+* The Renderer process **manages** only the corresponding web page. A crash in one Renderer process does not affect other Renderer processes.
+* The Renderer process **communicates** with the Main process via IPC to perform GUI operations in a web page. Calling native GUI-related APIs from the Renderer process directly is restricted due to security concerns and potential resource leakage.
+
+----
+
+The communication between processes is possible via Inter-Process Communication (IPC) modules: `ipcMain` and `ipcRenderer`.
+
+#### APIs
+
+##### Electron API
+
+Electron APIs are assigned based on the process type. It means that some of the methods can be used from the Main or Renderer processes respectively or both. API documentation indicates the process a particular API method can be used from.
+
+For example, to access Electron API in both processes, require its included module:
+
+```js
+const electron = require('electron')
+```
+
+To create a window, call the `BrowserWindow` class which is only available in the Main process:
+
+```js
+const { BrowserWindow } = require('electron')
+const win = new BrowserWindow()
+```
+
+To call the Main process from the Renderer, use the IPC module:
+
+```js
+// In the Main process
+const { ipcMain } = require('electron')
+
+ipcMain.handle('perform-action', (event, ...args) => {
+  // ... do actions on behalf of the Renderer
+})
+```
+
+```js
+// In the Renderer process
+const { ipcRenderer } = require('electron')
+
+ipcRenderer.invoke('perform-action', ...args)
+```
+
+> NOTE: Because the Renderer processes may run untrusted code (especially from third parties), it is important to carefully validate the requests that come to the Main process.
```

Comment:
```suggestion
> NOTE: Because Renderer processes may run untrusted code (especially from third parties), it is important to carefully validate the requests that come to the Main process.
```